### ISSUE TITLE
Build failure with GHC 9.2.1

### ISSUE THREAD

**Issue by sjakobi** [bos/bloomfilter#19]
```
Data/BloomFilter/Util.hs:43:38: error:
    • Couldn't match expected type ‘Word32#’ with actual type ‘Word#’
    • In the first argument of ‘W32#’, namely
        ‘(x# `uncheckedShiftL#` i#)’
      In the expression: W32# (x# `uncheckedShiftL#` i#)
      In an equation for ‘shiftL’:
          shiftL (W32# x#) (I# i#) = W32# (x# `uncheckedShiftL#` i#)
   |
43 |     shiftL (W32# x#) (I# i#) = W32# (x# `uncheckedShiftL#` i#)
   |                                      ^^^^^^^^^^^^^^^^^^^^^^^^

Data/BloomFilter/Util.hs:46:38: error:
    • Couldn't match expected type ‘Word32#’ with actual type ‘Word#’
    • In the first argument of ‘W32#’, namely
        ‘(x# `uncheckedShiftRL#` i#)’
      In the expression: W32# (x# `uncheckedShiftRL#` i#)
      In an equation for ‘shiftR’:
          shiftR (W32# x#) (I# i#) = W32# (x# `uncheckedShiftRL#` i#)
   |
46 |     shiftR (W32# x#) (I# i#) = W32# (x# `uncheckedShiftRL#` i#)
   |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^
cabal: Failed to build bloomfilter-2.0.1.0.
```

**Comment by sjakobi** [comment]
As a Hackage trustee, I have created a Hackage revision to prevent users from running into this build error: https://hackage.haskell.org/package/bloomfilter-2.0.1.0/revisions/

**Comment by sjakobi** [comment]
> As a Hackage trustee, I have created a Hackage revision to prevent users from running into this build error: https://hackage.haskell.org/package/bloomfilter-2.0.1.0/revisions/

Actually I had made a mistake in this revision. I had tightened the bound on `base` only for the testsuite.

I have now created a second revision that tightens the `base` bound for the library.

**Comment by joeyh** [comment]
fixed in 2.0.1.1
